### PR TITLE
[WARNING: SHITCODE]Introducing: The N-word Counter

### PR DESCRIPTION
**Pull request by BobJoga**
## About The Pull Request

(port of https://github.com/Skyrat-SS13/Skyrat13/pull/845 which will definitely be rejected)

Adds an N word counting subsystem, ~~that notifies admins whenever a player says a variation of the N word, and~~ which shows the total count of N words as well as all N word including messages in the round end report.
Future features may include:
Adding n word counts for each individual, and warning (or smiting, even) them if they exceed their n word pass treshold.
Adding the most n-word saying person to the round end report.
Counting Barack Obama (and variations) as 10 n words.
N word passes assigned to each player. Exceeding your assigned use of the n word pass results in smiting.

## Why It's Good For The Game

The gamer word is funny

## Changelog
:cl:
add: N word counter subsystem
tweak: Round end reports now show n word statistics 
/:cl:

### CODE REVIEW

File: code/modules/mob/say.dm [hunk 1]
```diff
@@ -5,6 +5,25 @@
 	if(GLOB.say_disabled)	//This is here to try to identify lag problems
 		to_chat(usr, "<span class='danger'>Speech is currently admin-disabled.</span>")
 		return
+	//HIPPIE CHANGES
+	if(findtext("nigger" || "nigga" || "n'wah" || "nations", message)) //N WORD COUNTER!
+		log_admin("[src] (Key: [src.key]) has said the n-word!")
```
Comment (KingTerrytheTerrible) [inline code/modules/mob/say.dm]: lmfao no

Comment (BobJoga) [reply code/modules/mob/say.dm]: the problem is that it will spam the logs right
doug4

Comment (yoyobatty) [reply code/modules/mob/say.dm]: yeah this will spam the logs

**Comment by BobJoga** [comment]
Since i've heard rumors that github tends to shut down PRs that have the gamer word (even if it is in the code on a fucking word filter or something), i've updated it to kind of sneakily do the funny word as a variable. Hopefully this will work and not fuck everything up.

**Comment by carlopault** [comment]
+1 hell yeah

**Comment by AnthonyJNgo** [comment]
Having certain words in code is fine as far as I am aware, as long as you are not posting them in comments/replies and as long as they make sense.

As an example, you wouldn't be able to write code that filters profanity if you didn't include the profanity in the code so it knows what to filter. 

With that said, why is this PR necessary? Seems a bit over the top honestly.

**Comment by Neo-0** [comment]
lmao

**Comment by BobJoga** [comment]
> Having certain words in code is fine as far as I am aware, as long as you are not posting them in comments/replies and as long as they make sense.
> 
> As an example, you wouldn't be able to write code that filters profanity if you didn't include the profanity in the code so it knows what to filter.
> 
> With that said, why is this PR necessary? Seems a bit over the top honestly.

It really isn't. It's a bad and stupid idea, which i think fits the "mood" of hippie well.

File: hippiestation.dme [hunk 2]
```diff
@@ -3724,4 +3725,5 @@
 #include "hippiestation\interface\skin.dmf"
 #include "interface\menu.dm"
 #include "interface\stylesheet.dm"
+#include "tools\Redirector\textprocs.dm"
```
Comment (Neo-0) [inline hippiestation.dme]: ^

**Comment by Timmeye** [comment]
+1 Frick admins and their heckin laws we want reddit gold

**Comment by karmaisblackandbluepilled** [comment]
make it work like the hall of glory from lifeweb
like, make it display an oppenable list of how many times everyone on the server said it for everyone
otherwise -4

**Comment by BobJoga** [comment]
> make it work like the hall of glory from lifeweb
> like, make it display an oppenable list of how many times everyone on the server said it for everyone
> otherwise -4

It already shows it in the round end report, just not in a very organized way

**Comment by ExpAntique** [comment]
+1 to the idea, -1 to the nazi-modding. remove the all the stuff about passes and smiting.

**Comment by cris5311** [comment]
-1
not cool

**Comment by BobJoga** [comment]
> +1 to the idea, -1 to the nazi-modding. remove the all the stuff about passes and smiting.

i didn't add those yet and i just kind of said those as a joke

**Comment by BobJoga** [comment]
> -1
> not cool

YOU are not cool

**Comment by MrEpicSax** [comment]
+1

**Comment by diegoxt45** [comment]
didn't you ban me from your server for having the n-word in my name

**Comment by Timmeye** [comment]
Inb4 someone makes a circuit that spams the N-word and it fills the round report up soo much that the server crashes or some wack stuff happens

**Comment by BobJoga** [comment]
> didn't you ban me from your server for having the n-word in my name

No, because the server wasn't mine.

File: code/modules/mob/say.dm [hunk 3]
```diff
@@ -5,6 +5,25 @@
 	if(GLOB.say_disabled)	//This is here to try to identify lag problems
 		to_chat(usr, "<span class='danger'>Speech is currently admin-disabled.</span>")
 		return
+	//HIPPIE CHANGES
+	var/nword1 = jointext(list("ni", "g", "g", "er"), "")
```
Comment (ghost) [inline code/modules/mob/say.dm:9]: why is this jointexted

Comment (Toxici11i) [reply code/modules/mob/say.dm:9]: github will ban you for having the entire word

Comment (ghost) [reply code/modules/mob/say.dm:9]: i think that's only in comments, though. not code.

btw, how do you think you got flagged?

Comment (Toxici11i) [reply code/modules/mob/say.dm:9]: nah, its incode too

Comment (BobJoga) [reply code/modules/mob/say.dm:9]: It wasn't jointext before, i dont know how github clowns work and i dont want to risk everything for saying the n word entirely even if its on code

**Comment by mkikt4743** [comment]
no, boss, that screenshot of me saying the n-word over sixteen thousand times does not represent my personality

**Comment by BobJoga** [comment]
Considering recent issues related to ToS compliance and such, i'll now be closing this repo and deleting the branch. I don't want to risk the integrity of the code as a whole.

**Pull request closed without merging**